### ISSUE TITLE
Names of generated proxy and service interfaces

### ISSUE THREAD

**Issue by bernardnormier** [icerpc/icerpc-csharp#2660]
In C#, for interface Hello, we currently generate:
 - interface IHelloProxy, an "invocation-side" abstraction
 - struct HelloProxy, which implements IHelloProxy and also the IProxy interface
 - interface IHello that users derive from when implementing a service that implements "Hello"

and then our examples use class Hello as the name for the example/typical service class that implement IHello.

This is all great except:
 - this convention won't work in any other language. This I-prefix for interfaces is unique to C#. For example, in Rust, we can't have trait HelloProxy implemented by struct HelloProxy
 - the only real _abstraction_ in this generated code is the invocation-side interface. That's the interface you can decorate. And that's the interface that application code actually calls. Remember that per DI principles, an abstraction is designed from the _client_ point of view. The dispatcher-side interface is not really an abstraction: application code doesn't / shouldn't call it. It's not really decoratable. The interface plays the role of a model for the service implementation.

So I think it would make sense to:
 - name the invocation-side abstraction after the interface name. For example interface Hello maps to invocation-side IHello
 - IHello is still implemented by struct HelloProxy, which also implements IProxy
   - "proxy" is really about an implementation of this abstraction: it's _an_ implementation that uses an invoker and a service address
 - on the dispatcher-side, we need a new name for this "model" interface. That's the difficulty! 
 - in our examples, we would implement it with class XxxService:
```
internal class HelloService : Service, IHello???
{

}
```

In summary, for Slice interface Hello:
 - invocation-side is generated IHello and HelloProxy
 - dispatcher-side is generated IHello??? + (in examples) HelloService
